src/scripts/core/models/button.ts

```diff
@@ -0,0 +1,88 @@
+import { Text as PixiText, Container, Graphics} from "pixi.js";
+import { gsap } from "gsap";
+
+export class Button extends Container {
+    private _buttonText: PixiText;
+    private _coin: Graphics;
+    private _buttonCallback: Function | undefined;
+
+    constructor (buttonText: string, buttonRadius: number, buttonCallback: Function) {
+        super();
+
+        this.interactive = true;
+        this.cursor = "pointer";
+
+        this.on('pointerdown',this.onPointerDown, this)
+        .on('pointerover', this.onPointerOver, this)
+        .on('pointerup', this.onPointerUp, this)
+        .on('pointerout', this.onPointerOut, this);
```

Comment:
Instead of naming the functions the same as the event which fires them, it would be better if the names matched what they were doing. This would make it easier for a dev to see whats happening on each event.

Reply:
If a function gets complex enough then naming it after what it does isn't going to work here. For some of them sure, but that's not a good practise to recommend tbh.

If we're talking personal preference too, I prefer these functions to be names after the event that's triggering them like right here. It's a preference thing really

Reply:
If the function is too complex to be named after what it does then the function needs breaking up into multiple functions. You should be able to name a function after its 'function' without worrying that there is more going on in the function. 
If you really wanted to name the function 'onPointerDown' then you could call another function inside that called 'tweenColour` - this would improve the readability even further.

Reply:
That would just add unnecessary layers to calls. But I will not request a change, I just wanted to add my 2 cents simply because this is a personal preference thing

Reply:
I can agree that its personal preference, as we don't have any rules or guidelines on this type of thing.
I agree its easy to see that 'onPointerDown' is happening on a pointerdown event, but it isn't clear what is actually happening on that event. 
`.on('pointerdown', this.changeCoinColour, this);` is so much easier to understand what is going on, and I can also look at the function name and know straight away what its doing, I don't have to look through the code to see what 'onPointerDown' actually does.

I haven't requested a change, I was just giving my thoughts and opinion, it's up to Kieran what he wants to do.